### DISCUSSION TITLE
How to support Parameters for a custom Jupyter kernel?

### DISCUSSION THREAD

**Discussion by hugetim** [quarto-dev/quarto-cli#11094] [Q&A] [answered]
### Description

The [Advanced > Jupyter Kernel Execution docs](https://quarto.org/docs/advanced/jupyter/kernel-execution.html#quarto-document-options) gave me the mistaken impression I could add support for Parameters to a Jupyter kernel by following the steps described there. But it's now becoming apparent that Jupyter Parameters uses Papermill under the hood, so getting Papermill to work with my language, Stata, is actually a prerequisite for supporting Quarto Parameters in a Stata kernel. Is that right?

If so, is it just a matter of adding Stata support to Papermill? Or are there likely to be other complications, do you think? I guess there's probably a reason the docs don't currently explain how to add Parameters support for a custom Jupyter kernel: it's complicated.

**Comment by cscheid** [comment, marked as answer]
I think it's "just" a matter of getting Papermill to work with your kernel, yes.

Or, at least, this is how we intended the behavior to work - up until now, we haven't had the chance to test the behavior in realistic settings. If things don't work out as you intend to, please reach out to us here or directly over email. Thanks!

**Reply by hugetim** [reply]
Ok, thanks. Yes, it seems that no changes to the kernel itself were needed to get Parameters to work. I have it working now, given a modified version of Papermill that supports Stata: https://github.com/nteract/papermill/pull/809/files

**Reply by cscheid** [reply]
It's a bit of a bummer that Papermill needs to be changed to support new programming languages, but I'm glad you sorted it out!
